model/variable.cc

```diff
@@ -209,9 +209,21 @@ void VariableBase::ensureValueExists(VariableValue* vv, const std::string& nm) c
 string VariableBase::init() const
 {
   auto value=minsky().variableValues.find(valueId());
-  if (value!=minsky().variableValues.end())
+  if (value!=minsky().variableValues.end()) {
+    //if (auto i=dynamic_cast<IntOp*>(controller.lock().get()))
+    //  if (i->ports.size()>1 && !i->ports[1]->wires().empty())
+    //     i->intVar->init(i->ports[1]->wires()[0]->from()->item().variableCast()->init());
+    //else if (auto g=dynamic_cast<GodleyIcon*>(controller.lock().get()))
+    //  for (auto v: g->stockVars())
+    //     if (v->ports.size()>1 && !v->ports[1]->wires().empty())
+    //        v->init(v->ports[1]->wires()[0]->from()->item().variableCast()->init());        	  
+	if (inputWired())
+        value->second->init=ports[1]->wires()[0]->from()->item().variableCast()->init();
```

Comment:
Not sure what you're trying to do here, but the item the wire is from needn't be a variable, so potentially you will be dereferencing a null pointer.

Truth is, only integral variables need their initialisation updated according to what they're attached to, and I think that might be handled elsewhere.

Reply:
> Not sure what you're trying to do here, but the item the wire is from needn't be a variable, so potentially you will be dereferencing a null pointer.

I took inspiration from how vars and pars inherit their units from input wires and set their initial values to the value of the par or var on the left hand side.

> Truth is, only integral variables need their initialisation updated according to what they're attached to, and I think that might be handled elsewhere.

Kresimir also pointed out that stock vars in Godleys are not initialised correctly if their initial value is a symbolic par or var that, in turn, inherits their intial value from a another par or var via a wire.